src/rust/src/params.rs

```diff
@@ -0,0 +1,1440 @@
+}
+
+impl FromStr for &str {
+    fn to_ms(&self) -> CcxBoundaryTime {
+        let mut parts = self.rsplit(":");
+
+        let seconds: u32;
+
+        match parts.next() {
+            Some(sec) => seconds = sec.parse().unwrap(),
+            None => {
+                println!("Malformed timecode: {}", self);
+                std::process::exit(ExitCode::MalformedParameter as i32);
+            }
+        };
+        let mut minutes: u32 = 0;
+
+        match parts.next() {
+            Some(mins) => minutes = mins.parse().unwrap(),
+            None => {}
+        };
+        let mut hours: u32 = 0;
+
+        match parts.next() {
+            Some(hrs) => hours = hrs.parse().unwrap(),
+            None => {}
+        };
+
+        if seconds > 60 || minutes > 60 {
+            println!("Malformed timecode: {}", self);
+            std::process::exit(ExitCode::MalformedParameter as i32);
+        }
+
+        CcxBoundaryTime {
+            hh: hours,
+            mm: minutes,
+            ss: seconds,
+            time_in_ms: hours + 60 * minutes + 3600 * seconds,
+            set: true,
+        }
+    }
+}
+
+fn get_vector_words(string_array: &[&str]) -> Vec<String> {
+    let mut vector = Vec::new();
+    for string in string_array {
+        vector.push(String::from(*string));
+    }
+    return vector;
+}
+
+fn atoi_hex<T>(s: &str) -> Result<T, &str>
+where
+    T: Integer + std::str::FromStr,
+{
+    if s.len() > 2 && s.to_lowercase().starts_with("0x") {
+        // Hexadecimal
+        T::from_str_radix(&s[2..], 16).map_err(|_| "not a valid hexadecimal number")
+    } else {
+        // Decimal
+        s.parse::<T>().map_err(|_| "not a valid decimal number")
+    }
+}
+
+fn get_atoi_hex<T>(s: &str) -> T
+where
+    T: Integer + std::str::FromStr<Err = std::num::ParseIntError>,
+{
+    match atoi_hex(s) {
+        Ok(val) => val,
+        Err(_) => {
+            println!("Malformed parameter: {}", s);
+            std::process::exit(ExitCode::MalformedParameter as i32)
+        }
+    }
+}
+
+fn process_word_file(filename: &str, list: &mut Vec<String>) -> Result<(), std::io::Error> {
+    let file = File::open(filename)?;
+    let reader = BufReader::new(file);
+    let mut num = 0;
+
+    for line in reader.lines() {
+        num += 1;
+        let line = line.unwrap();
+        if line.starts_with('#') {
+            continue;
+        }
+
+        let new_len = line.trim().len();
+        if new_len > CCX_DECODER_608_SCREEN_WIDTH as usize {
+            println!(
+                "Word in line {} too long, max = {} characters.",
+                num, CCX_DECODER_608_SCREEN_WIDTH
+            );
+            continue;
+        }
+
+        if new_len > 0 {
+            list.push(line.trim().to_string());
+        }
+    }
+    Ok(())
+}
+
+fn set_output_format_bin(opt: &mut CcxSOptions) {
+    opt.write_format = CcxOutputFormat::Rcwt;
+}
+
+fn set_output_format(opt: &mut CcxSOptions, args: &Args) {
```

Comment:
Every branch in this function requires `args.out` to be `Some`. So you could write it as follows:-
```
if let Some(out) = args.out {
	// if else ladder here
	// use (out == OutFormat::) instead of (can_unwrap && args.out.unwrap() == OutFormat::)
} else {
	// Error part here
}

Reply:
Fixed, although I have used something different but end result is same.